src/ui/Analytics/Analytics.ts

```diff
@@ -158,7 +162,69 @@ export class Analytics extends Component {
+     * - Set this option to `true` to ensure that the component populates the data layer even if a GTM container snippet cannot be found in the page.
+     *
+     * See also [`gtmDataLayerName`]{@link Analytics.options.gtmDataLayerName}.
+     *
+     * By default, if a standard GTM container snippet is found in the page, the component automatically attempts to push to its initialized data layer (see [Google Tag Manager Quick Start Guide](https://developers.google.com/tag-manager/quickstart)).
+     */
+    pushToGtmDataLayer: ComponentOptions.buildBooleanOption({
+      postProcessing: (value?: boolean, options?: IAnalyticsOptions) => {
+        const logger = new Logger('Analytics');
+        let optionValue = false;
+        if (typeof value === 'boolean') {
+          optionValue = value;
+        } else {
+          const includesGtmScript = new RegExp(`gtm\\.js\\?id='\\+i\\+dl;f\\.parentNode.insertBefore\\(j,f\\);`);
+          const initializesGtmScript = new RegExp(`\\(window,document,'script','(\\w+)','GTM-[A-Z0-9]{7}'\\)`);
+          const gtmContainerSnippet = _.find(document.querySelectorAll('script'), scriptElement => {
+            // Could yield false positives (e.g., if the innerText is commented out).
+            return includesGtmScript.test(scriptElement.innerText) && initializesGtmScript.test(scriptElement.innerText);
+          });
+          if (gtmContainerSnippet) {
+            const match = initializesGtmScript.exec(gtmContainerSnippet.innerText);
+            // Attempt to extract data layer name from GTM container snippet
+            if (typeof match[1] === 'string') {
+              options.gtmDataLayerName = match[1];
+            }
+            optionValue = true;
+            logger.info(
+              `Standard Google Tag Manager container snippet found in page; automatically attempting to push to data layer '${
+                options.gtmDataLayerName
+              }'. Explicitly set the 'pushToGtmDataLayer' option of the 'Analytics' component to 'false' if you do not wish to push Coveo usage analytics events to the GTM data layer.`
+            );
+          }
+        }
+        return optionValue as boolean;
+      }
+    }),
+
+    /**
+     * The name of the Google Tag Manager data layer initialized in the page.
+     *
+     * **Note:**
+     * Setting this option is only useful if [`pushToGtmDataLayer`]{@link Analytics.options.pushToGtmDataLayer} is explicitly set to `true`, and the [GTM data layer was renamed](https://developers.google.com/tag-manager/devguide#renaming) in the page.
+     *
+     * By default, if `pushToGtmDatalayer` is left `undefined` and a GTM container snippet is found in the page, the component attempts to extract the data layer name from the snippet. If it fails to do so, or if `pushToGtmDataLayer` is explicitly set to `true`, this option defaults to `dataLayer`.
+     */
+    gtmDataLayerName: ComponentOptions.buildStringOption({
```

Comment:
In line with the above, I think we should remove the default name `dataLayer` and force people to set it if they want to push to Google analytics.